electrumsv/wallet.py

```diff
@@ -977,8 +1032,27 @@ def get_address_history(self, address: Address):
                 if tx_fee is not None:
                     flags |= TxFlags.HasFee
                 updates.append((tx_hash, data, None, flags))
+
+            # If HasByteData --> StateCleared
+            # fast-tracks Signed, Dispatched, Received transactions
+            entry = self._datastore.tx.get_entry(tx_hash)
             self._datastore.tx.update_or_add(updates)
 
+            if entry and (entry.flags & TxFlags.HasByteData) and not \
```

Comment:
This if clause needs to be merged correctly with the loop that follows it. Also it should not be calling add_transaction. I suspect the following would suffice.
```
            for tx_id in set(t[0] for t in hist):
                if self._datastore.tx.have_transaction_data(tx_id):
                    entry = self._datastore.tx.get_entry(tx_id)
                    # if addr is new, we have to recompute txi and txo
                    if not len(self.get_txins(tx_id, address)) and \
                       not len(self.get_txouts(tx_id, address)):
                        self.apply_transactions_xputs(tx_id, entry.transaction)
                   if entry.flags & (TxFlags.StateCleared | TxFlags.StateSettled) == 0:
                       self.set_transaction_state(tx_id, TxFlags.StateCleared | TxFlags.HasBytedata) 
                       ... handle paments...
```

Reply:
Nice :wink: pretty much worked with minimal modification.